triplestore-jena-memory/src/main/java/zone/cogni/asquare/triplestore/jenamemory/LocalTdbRdfStoreService.java

```diff
@@ -89,7 +85,7 @@ public LocalTdbRdfStoreService(
 
     this.dataset = TDBFactory.createDataset(tdbLocationFolder.getAbsolutePath());
     initialize(initFolder);
-    this.ready = true;
+    this.ready.set(true);
```

Comment:
if this is true when it does not fail, should we not have a finally for when it fails? maybe to do some cleanup related to what is setup during execution of the constructor?

Reply:
I didn't like the idea that we initialize TDB in the constructor but I don't want to change it because then we'd break backward compatibility.
My idea was to follow the original idea of the `LocalTdbRdfStoreService` where if a problem happens then it is not handled locally but propagated back on the caller chain, hence the `LocalTdbRdfStoreService` instance is not even constructed. If I'd introduce a new state where we handle issues locally and say may LocalTdbRdfStoreService instance exists but does not work well then we will need to re-work all the already existing codes to handle this new situation.
The `ready` is the same state as the `close` state was on the previous implementation, it's used the same way, just with opposite meaning

Reply:
Cleanup would be nice tho' but I'm not sure if this is the best place to handle it because there can be a lot of issues and it's hard to now how to go back to a consistent state from here.

Reply:
What I started to do is that I use LocalTdbRdfStoreService instances through the pool and when one instance creation fails then it is invalidated by the pool and on the application I listen for different type of issues during invalidation and see if we can react on them. For example if there is TDB corruption then we can react on it by disabling access to this TDB from other threads, put TDB into maintenance mode (this is a new mode on the app), export the content and re-create the TDB from export

Reply:
ah, no. I fully agree that we should rethrow the issue. so, probably having a catch where we would cleanup resources (I dunno if there are any...) like connections, files etc could help. but indeed, construction should not be a "success" but should still throw an error.

so, probably not finally but catch with cleanup and rethrow.